src/Agent.Worker/JobExtension.cs

```diff
@@ -691,6 +695,23 @@ public async Task FinalizeJob(IExecutionContext jobContext)
             return snapshot;
         }
 
+        private void OutputImageVersion(IExecutionContext context)
+        {
+            string imageVersion = System.Environment.GetEnvironmentVariable(Constants.ImageVersionVariable);
+
+            if (imageVersion != null)
```

Comment:
Is this expected that sometimes such env variable is not present?
Maybe make sense to trace such case, or publish telemetry

Reply:
Thanks! Added tracing for when the variable is unset and telemetry for all cases